cpp/src/parquet/statistics.cc

```diff
@@ -610,8 +619,12 @@ class TypedStatisticsImpl : public TypedStatistics<DType> {
     if (HasNullCount()) {
       s.set_null_count(this->null_count());
     }
-    // num_values_ is reliable and it means number of non-null values.
-    s.all_null_value = num_values_ == 0;
+    if (HasNullCount()) {
+      // num_values_ is reliable and it means number of non-null values.
+      s.all_null_value = num_values_ == 0;
```

Comment:
Can you add a statistics test for `all_null_value`? Including cases where `num_values_` is not reliable.

Reply:
Added